app/src/main/java/com/ggetters/app/ui/central/viewmodels/HomeProfileViewModel.kt

```
@@ -59,15 +59,29 @@ class HomeProfileViewModel @Inject constructor(
-            userRepository.upsert(player)
-            _player.value = player
+            try {
+                userRepository.upsert(player)
+                // Sync changes to remote
+                userRepository.sync()
+                _player.value = player
+            } catch (e: Exception) {
+                // Handle error - could show error message to user
+                _player.value = null
+            }
         }
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)